src/QuickCheckVEngine/RVFI_DII.hs

```diff
@@ -37,62 +38,123 @@
 -- SUCH DAMAGE.
 --
 
-{-|
-    Module      : QuickCheckVEngine.RVFI_DII
-    Description : The RVFI-DII interface
-
-    This module re-exports the 'QuickCheckVEngine.RVFI_DII.RVFI' and
-    'QuickCheckVEngine.RVFI_DII.DII' modules, and provides functions to send and
-    receive 'DII_Packet's and 'RVFI_Packet's over a 'Socket'.
--}
-
-module QuickCheckVEngine.RVFI_DII (
-  module QuickCheckVEngine.RVFI_DII.RVFI
-, module QuickCheckVEngine.RVFI_DII.DII
-, sendDIIPacket
-, sendDIITrace
-, recvRVFIPacket
-, recvRVFITrace
-) where
-
-import QuickCheckVEngine.RVFI_DII.RVFI
-import QuickCheckVEngine.RVFI_DII.DII
+-- |
+--    Module      : QuickCheckVEngine.RVFI_DII
+--    Description : The RVFI-DII interface
+--
+--    This module re-exports the 'QuickCheckVEngine.RVFI_DII.RVFI' and
+--    'QuickCheckVEngine.RVFI_DII.DII' modules, and provides functions to send and
+--    receive 'DII_Packet's and 'RVFI_Packet's over a 'Socket'.
+module QuickCheckVEngine.RVFI_DII
+  ( module QuickCheckVEngine.RVFI_DII.RVFI,
+    module QuickCheckVEngine.RVFI_DII.DII,
+    RvfiDiiConnection(..),
+    sendDIIPacket,
+    sendDIITrace,
+    recvRVFITrace,
+    rvfiNegotiateVersion,
+  )
+where
 
-import Data.Int
-import Data.Binary
 import Control.Monad
+import Data.Binary
+import Data.Binary.Get
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.ByteString.Lazy.Char8 as C8
+import Data.Int
 import Network.Socket
 import Network.Socket.ByteString.Lazy
-import qualified Data.ByteString.Lazy as BS
+import QuickCheckVEngine.RVFI_DII.DII
+import QuickCheckVEngine.RVFI_DII.RVFI
+
+data RvfiDiiConnection = RvfiDiiConnection
+  { socket :: Socket,
+    traceVersion :: Int,
+    name :: String,
+    verbosity :: Int
+  }
```

Comment:
Why make it a record when it's used like a normal data type? (Or alternatively "why use it like a normal data type when it's a record?")

Reply:
Because I'm terrible at Haskell and don't know the difference 😢

Reply:
Hopefully resolved now?